### PR TITLE
Use new MacOS arm64 pool for official builds

### PR DESCRIPTION
**Pull request by steveisok**
This pool is only available internally for now, so let's use it!

### CODE REVIEW

**Comment by steveisok** [comment]
Official build run https://dev.azure.com/dnceng/internal/_build/results?buildId=2372706&view=results

File: eng/pipelines/common/xplat-setup.yml [hunk 1]
```diff
@@ -171,9 +171,13 @@ jobs:
           demands: ImageOverride -equals Build.Ubuntu.2204.Amd64
 
         # OSX Build Pool (we don't have on-prem OSX BuildPool).
-        ${{ if in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator') }}:
+        ${{ if and(in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator'), eq(variables['System.TeamProject'], 'public')) }}:
           vmImage: 'macos-12'
 
+        # Official build OSX pool
+        ${{ if and(in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator'), ne(variables['System.TeamProject'], 'public')) }}:
+          vmImage: 'macos-13-arm64'
```
Comment (jkotas) [inline eng/pipelines/common/xplat-setup.yml:179]: We should double that these builds use native Arm64 SDK after the switch.

(x64 SDK running on Arm64 emulator is slow and it has reliability issues.)

**Comment by filipnavara** [comment]
I assume there’s a newer Xcode on these machines, right? It breaks NativeAOT builds.

**Comment by jkotas** [comment]
These machines have Xcode_15.0.1.

**Comment by filipnavara** [comment]
Issue: https://github.com/dotnet/runtime/issues/97745#issuecomment-1925915381
Workaround: https://github.com/dotnet/runtime/pull/97960

Note that just applying the workaround would not be enough since the tools are built with published ILCompiler, not the in-tree one.

**Comment by steveisok** [comment]
@jkotas @filipnavara can we split building ILCompiler / nativeaot runtime packs into its own leg that uses the old macs?

**Comment by filipnavara** [comment]
> can we split building ILCompiler / nativeaot runtime packs into its own leg that uses the old macs?

There’s a PR to use old linker for net8.0 branch (https://github.com/dotnet/runtime/pull/97856). We can probably take the bits to unblock this PR. Also, there we need to check https://github.com/dotnet/runtime/issues/98124. I can do both tomorrow if you feel like it’s worth pushing it through.

**Comment by jkotas** [comment]
I agree with @filipnavara that applying the workaround is the right way to go. Splitting the native aot specific builds into its own legs would be non-trivial and it would be colliding with what we are trying to do for Unified Build.

**Comment by akoeplinger** [comment]
FYI I've seen a couple random disconnects with this pool in the VMR build: https://dev.azure.com/dnceng/internal/_build?definitionId=1330&_a=summary&branchFilter=256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180%2C256180&statusFilter=cancelled

Interestingly all of them in the "Publish BuildLogs" step which just uploads files to AzDO...
I think we have some contact on the Mac pool team we could ping right?

**Comment by steveisok** [comment]
> Interestingly all of them in the "Publish BuildLogs" step which just uploads files to AzDO... I think we have some contact on the Mac pool team we could ping right?

I'd post on FR if you didn't already.

**Review by jkotas — APPROVED** [review]

**Review by hoyosjs — APPROVED** [review]

**Comment by steveisok** [comment]
> Have we done multiple successful trial runs on the new pool? If yes, this looks good to me.

Yes, I kicked it off a few times.

**Review by filipnavara — APPROVED** [review]

**Review by agocke — APPROVED** [review]

**Pull request merged**